### ISSUE TITLE
ClickHouse: ORDER BY ... WITH FILL ... INTERPOLATE (...) is unparsable

### ISSUE THREAD

**Issue by friedrich123456789** [sqlfluff/sqlfluff#8029]
### What happened

The `INTERPOLATE (...)` clause of `ORDER BY ... WITH FILL` is unparsable.
`WITH FILL` on its own parses fine.

### Expected behaviour

`ORDER BY <expr> WITH FILL INTERPOLATE ( <expr> [AS <expr>], ... )` parses.

### Observed behaviour

```
L1:P38 PRS: Found unparsable section: 'INTERPOLATE (b AS b + 1)'
```

### How to reproduce

```sql
SELECT a FROM t ORDER BY a WITH FILL INTERPOLATE (b AS b + 1)
```

```bash
sqlfluff parse --dialect clickhouse -
```

Valid, verified on ClickHouse 26.4.2.
Docs: https://clickhouse.com/docs/sql-reference/statements/select/order-by
(the `WITH FILL` / `INTERPOLATE` modifier).

### Dialect

clickhouse

### Version

sqlfluff 4.2.2 (latest release), Python 3.12; also applies to `main`.

### Configuration

None — default rules, `--dialect clickhouse`.
